### ISSUE TITLE
Unsubscribing from handlers, multiple handlers

### ISSUE THREAD

**Issue by TheRusskiy** [OneSignal/react-native-onesignal#1264]
**Description:**
Before the version 4.0.0 you could use the following pattern, e.g. in hooks:
```
    useEffect(() => {
      const onReceived = () => {...}
      const onOpened = () => {...}
      OneSignal.addEventListener("received", onReceived)
      OneSignal.addEventListener("opened", onOpened)
      return () => {
        OneSignal.removeEventListener("received", onReceived)
        OneSignal.removeEventListener("opened", onOpened)
      }
   }, [])
```
After version 4 you can only set a handler, without being able to remove it.
Moreover, you can only set it once, all the previous calls are overridden (looking at code).

**Comment by jkasten2** [comment]
@TheRusskiy Thanks for the feedback.
The "received" handler has changed significantly in 4.0.0, this is now [`OneSignal.setNotificationWillShowInForegroundHandler`](https://documentation.onesignal.com/docs/sdk-notification-event-handlers#foreground-notification-received-event). This requires you to call `complete()` on the event so it no longer is well suited to multiple receivers.
The "open" event which is now [`setNotificationOpenedHandler`](https://documentation.onesignal.com/docs/sdk-notification-event-handlers#foreground-notification-received-event) was moved to a setter for consistently.

Could I ask the use cases you have more multiple receivers? We can take this extra information into consideration when we making API changes.

In mean time would be to setup these handlers when your app starts and then broadcast them to the parts of your app that need it. Not as ideal as OneSignal handling this logic for you but should unblock you from upgrading to the 4.0.0 version.

**Comment by rgomezp** [comment]
Closing due to inactivity

**Comment by TheRusskiy** [comment]
@rgomezp @jkasten2 sorry for the lack of response, I have postponed OneSignal update so didn't have an opportunity to try the suggested changes.
If there are issues still I am going to open a new one.
Thanks for the advice!
